### ISSUE TITLE
Template evaluation time limit too short

### ISSUE THREAD

**Issue by ihatetoregister** [ridiculousfish/HexFiend#157]
When fiddling around with templates I reached the time limit for the template evaluation. I'm not sure it's something wrong with my template or if my old MacBook Air is just too slow. Currently, it is set to 2 seconds but I suggest increasing it to at least 10 seconds. 

Test template: https://gist.github.com/ihatetoregister/a20294daae6093b23d0a5c6b4a44b1fc
Input file: https://twisted-electrons.com/downloads/TherapSid-Factory1.zip

I've tested increasing the time in the evaluateScript function from 2 to 10 and it seem to work. 

```
- (NSString *)evaluateScript:(NSString *)path {
    Tcl_LimitTypeSet(_interp, TCL_LIMIT_TIME);
    Tcl_Time time;
    Tcl_GetTime(&time);
    time.sec += 10; // Changed from 2 to 10
    Tcl_LimitSetTime(_interp, &time);
    const int err = Tcl_EvalFile(_interp, [path fileSystemRepresentation]);
    if (err != TCL_OK) {
        Tcl_Obj *options = Tcl_GetReturnOptions(_interp, err);
        Tcl_Obj *key = Tcl_NewStringObj("-errorinfo", -1);
        Tcl_Obj *value = NULL;
        Tcl_IncrRefCount(key);
        Tcl_DictObjGet(NULL, options, key, &value);
        Tcl_DecrRefCount(key);
        if (value) {
            return [NSString stringWithUTF8String:Tcl_GetStringFromObj(value, NULL)];
        }
    }
    return nil;
}
```

**Comment by bvandergiessen** [comment]
This might be a good idea. Tcl doesn't seem the fastest of languages.

**Comment by kainjow** [comment]
Probably would make sense to make this a preference.

**Comment by kainjow** [comment]
I bumped default to 10 and allowed it to be overridden. Currently needs to be done via the command line. For example to change it to 30:

    defaults write com.ridiculousfish.HexFiend BinaryTemplateScriptTimeout -int 30

**Comment by xyzzy-plugh-plover** [comment]
This worked in 2.15, but doesn't seem to work in 2.16. Is there a new setting?

**Comment by kainjow** [comment]
Seems to be working fine for me. Can you provide more details?

**Comment by xyzzy-plugh-plover** [comment]
I ran `defaults write com.ridiculousfish.HexFiend BinaryTemplateScriptTimeout -int 30000` on 2.15; templates take a long time, but finish.

I ran `defaults write com.ridiculousfish.HexFiend BinaryTemplateScriptTimeout -int 30000` on 2.16; times out after 10 seconds.

I've tried higher & lower numbers, but 2.16 always times out after 10 seconds.

`defaults write com.ridiculousfish.HexFiend BinaryTemplateScriptTimeout` always returns the correct number that I most recently input.

Has the domain and/or key changed?

Can this be configured via the GUI now?

Thanks.

**Comment by xyzzy-plugh-plover** [comment]
@kainjow Any thoughts about why `com.ridiculousfish.HexFiend BinaryTemplateScriptTimeout` isn't working for me in 2.16?

I'm running macOS 12.6.8 on Intel.

I previously had HexFiend installed via the App Store, but I uninstalled it when I realized that Homebrew had 2.16 available from but the App Store only had 2.15.

Maybe Homebrew is at fault, or maybe my uninstalled App Store version got in the way.
